src/dns/query.hpp

```diff
@@ -203,13 +203,17 @@ static void dns_callback(int code, char type, int count, int ttl,
 
     context->message.answers =
             build_answers_evdns(code, type, count, ttl, addresses);
-    if (context->message.error_code != DNS_ERR_NONE) {
-        context->callback(mk::dns::dns_error(context->message.error_code),
-                context->message);
-    } else {
-        context->callback(NoError(), context->message);
+    try {
+        if (context->message.error_code != DNS_ERR_NONE) {
+            context->callback(mk::dns::dns_error(context->message.error_code),
+                    context->message);
+        } else {
+            context->callback(NoError(), context->message);
+        }
+    } catch (const Error& e) {
```

Comment:
Good for [using const](http://stackoverflow.com/questions/2145147/why-catch-an-exception-as-reference-to-const)! We should use it in more places!